From 0cf6ee4835d4adb9fdfab8115117cb68a62c849a Mon Sep 17 00:00:00 2001
From: Yao Zi <ziyao@disroot.org>
Date: Fri, 27 Jun 2025 06:44:05 +0000
Subject: [PATCH] tools: Compile maglev files for v8_compiler only when it's
 disabled

v8_compiler needs maglev graph builder even if maglev is disabled. But
the files shouldn't be compiled for libv8_internal_headers, which is
required by both v8_base_without_compiler and v8_compiler, causing
duplicate symbols when building with maglev disabled.

Signed-off-by: Yao Zi <ziyao@disroot.org>
---
 tools/v8_gypfiles/v8.gyp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/tools/v8_gypfiles/v8.gyp b/tools/v8_gypfiles/v8.gyp
index 0169cb0..d8676c5 100644
--- a/tools/v8_gypfiles/v8.gyp
+++ b/tools/v8_gypfiles/v8.gyp
@@ -633,11 +633,6 @@
           '<!@pymod_do_main(GN-scraper "<(V8_ROOT)/BUILD.gn"  "v8_header_set.\\"v8_internal_headers\\".*?sources = ")',
         ],
         'conditions': [
-          ['v8_enable_maglev==0', {
-            'sources': [
-              '<!@pymod_do_main(GN-scraper "<(V8_ROOT)/BUILD.gn"  "v8_header_set.\\"v8_internal_headers\\".*?!v8_enable_maglev.*?sources \\+= ")',
-            ],
-          }],
           ['v8_enable_snapshot_compression==1', {
             'sources': [
               '<!@pymod_do_main(GN-scraper "<(V8_ROOT)/BUILD.gn"  "v8_header_set.\\"v8_internal_headers\\".*?v8_enable_snapshot_compression.*?sources \\+= ")',
@@ -953,6 +948,11 @@
         'abseil.gyp:abseil',
       ],
       'conditions': [
+        ['v8_enable_maglev==0', {
+          'sources': [
+            '<!@pymod_do_main(GN-scraper "<(V8_ROOT)/BUILD.gn"  "v8_header_set.\\"v8_internal_headers\\".*?!v8_enable_maglev.*?sources \\+= ")',
+          ],
+        }],
         ['v8_enable_turbofan==1', {
           'dependencies': ['v8_compiler_sources'],
         }, {
-- 
2.50.0

